workbench/src/renderer/components/PluginModal/index.jsx

```diff
@@ -34,15 +34,62 @@ export default function PluginModal(props) {
   const [needsMSVC, setNeedsMSVC] = useState(false);
   const [plugins, setPlugins] = useState({});
   const [installFrom, setInstallFrom] = useState('url');
+  const [userAcknowledgment, setUserAcknowledgment] = useState(false);
+  const [userAcknowledgmentError, setUserAcknowledgmentError] = useState(false);
+  const [pluginSourceMissingError, setPluginSourceMissingError] = useState(false);
 
   const handleModalClose = () => {
     setURL('');
     setRevision('');
     setInstallErr('');
     setUninstallErr('');
+    clearFormErrors();
     closeModal();
   };
 
+  const clearFormErrors = () => {
+    setUserAcknowledgmentError(false);
+    setPluginSourceMissingError(false);
+  };
+
+  useEffect(() => {
+    clearFormErrors();
+  }, [installFrom]);
+
+  useEffect(() => {
+    if (pluginSourceMissingError) {
+      setPluginSourceMissingError(false);
+    }
+  }, [url, path]);
+
+  useEffect(() => {
+    if (userAcknowledgment) {
+      setUserAcknowledgmentError(false);
+    }
+  }, [userAcknowledgment]);
+
+  const handleAddPluginClick = () => {
+    clearFormErrors();
+    if (validateAddPluginForm()) {
+      addPlugin();
+    }
+  };
+
+  const validateAddPluginForm = () => {
+    let formValid = true;
+    if ((installFrom === 'url' && !url)
+        || (installFrom === 'path' && !path)
```

Comment:
I'm not sure about needing to show this warning when installing from a local path. I don't know much about security, but it seems that if the malicious code is already downloaded onto someone's computer, it may be too late? And the more likely case is that someone is installing a plugin that they've developed themselves